cf-console-header-menu-button-sample/src/aem-cf-console-admin-1/web-src/src/components/InsightsModal.js

```diff
@@ -0,0 +1,148 @@
+  }
+  const forecast = await resp.json();
+  const data = forecast?.properties?.periods?.[0] ?? null;
+  if (data !== null) {
+    data.city = city;
+  }
+  return data;
+};
+
+export default InsightsModal = () => {
+  const [guestConnection, setGuestConnection] = useState();
+
+  const [weatherForecast, setWeatherForecast] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      const guestConnection = await attach({ id: extensionId });
+
+      setGuestConnection(guestConnection);
+    })();
+  }, []);
+
+  const onActionBtnHandler = async (city) => {
+    setIsLoading(true);
+    try {
+      const result = await getWeatherForecast(city);
+      setWeatherForecast(result);
+      displayToast(`Weather forecast for ${city} has been successfully fetched.`);
+    } catch (error) {
+      console.error("error", error);
+      displayToast(`Error fetching weather forecast for ${city}`, false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const onCloseHandler = () => {
+    guestConnection.host.modal.close();
+  };
+
+  const displayToast = useCallback((toast, successful = true) => {
+    guestConnection.host.toaster.display({
+      variant: successful ? "primary" : "negative",
+      message: toast,
+      timeout: 1500,
+    });
+  }, [guestConnection]);
+
+  return (
+    <Provider theme={defaultTheme} colorScheme="light">
+      <Content width="100%">
+        {isLoading
+          ? (
+            <Spinner />
+          )
+          : (
+            <>
+              <View marginBottom="size-100">
+                <MenuTrigger>
+                  <ActionButton>
+                    Display Weather Forecast
+                  </ActionButton>
+                  <Menu onAction={onActionBtnHandler}>
+                    <Item key={AUSTIN}>{AUSTIN}</Item>
+                    <Item key={NEW_YORK}>{NEW_YORK}</Item>
+                    <Item key={LOS_ANGELES}>{LOS_ANGELES}</Item>
+                  </Menu>
+                </MenuTrigger>
+              </View>
+
+              {(weatherForecast !== null) &&
+                <Flex height="size-3000" gap="size-100">
+                  <View flexBasis="20%" flexGrow="0" flexShrink="0">
+                    <img src={weatherForecast.icon} className="weather-forecast-img" />
+                  </View>
+                  <View UNSAFE_className="weather-forecast-description">
+                    <Heading level={2} marginTop="0">{weatherForecast.city}</Heading>
+                    <p>
+                      Temperature: {weatherForecast.temperature}°{weatherForecast.temperatureUnit} /{" "}
+                      {Math.round(((weatherForecast.temperature - 32) * 50) / 9) / 10}°C
+                    </p>
+                    <p>{weatherForecast.shortForecast}</p>
+                    <p>{weatherForecast.detailedForecast}</p>
+                    <p>{new Date(weatherForecast.startTime).toLocaleString()}</p>
```

Comment:
Let's use [LabeledValue](https://react-spectrum.adobe.com/react-spectrum/LabeledValue.html#dates-and-times) for the date and move it next to the city name.